st2common/tests/unit/services/test_packs.py

```diff
@@ -0,0 +1,193 @@
+
+        # creating only test entry_point file in dummy pack
+        with open(entry_point, "w") as f:
+            f.write("# entry point file to be removed")
+
+        # asserting metadata file doesn't exists
+        self.assertFalse(os.path.exists(metadata_file))
+
+        # asserting entry_point files exists
+        self.assertTrue(os.path.exists(entry_point))
+
+        delete_action_files_from_pack(TEST_PACK, entry_point, metadata_file)
+
+        # asserting both entry_point and metadata files doesn't exists
+        self.assertFalse(os.path.exists(entry_point))
+        self.assertFalse(os.path.exists(metadata_file))
+
+
+class DeleteActionFilesErrorTest(unittest2.TestCase):
+    def setUp(self):
+        entry_point = os.path.join(TEST_PACK_PATH, "actions", "test_entry_point.py")
+        metadata_file = os.path.join(TEST_PACK_PATH, "actions", "test_metadata.yaml")
+
+        # creating test entry_point file in dummy pack
+        with open(entry_point, "w") as f:
+            f.write("# entry point file to be removed")
+
+        # creating test metadata file in dummy pack
+        with open(metadata_file, "w") as f:
+            f.write("# metadata file to be removed")
+
+    def tearDown(self):
+        entry_point = os.path.join(TEST_PACK_PATH, "actions", "test_entry_point.py")
+        metadata_file = os.path.join(TEST_PACK_PATH, "actions", "test_metadata.yaml")
+
+        # removing test entry_point and metadata file in dummy pack
+        os.remove(entry_point)
+        os.remove(metadata_file)
+
+    @mock.patch.object(os, "remove")
+    def test_permission_error_to_remove_resource_entry_point_file(self, remove):
+
+        entry_point = os.path.join(TEST_PACK_PATH, "actions", "test_entry_point.py")
+        metadata_file = os.path.join(TEST_PACK_PATH, "actions", "test_metadata.yaml")
+
+        remove.side_effect = PermissionError("No permission to delete file from disk")
+
+        # asserting metadata files exists
+        self.assertTrue(os.path.exists(entry_point))
+
+        # asserting delete_action_files_from_pack function raises PermissionError
+        with self.assertRaises(PermissionError):
+            delete_action_files_from_pack(TEST_PACK, entry_point, metadata_file)
+
+    @mock.patch.object(os, "remove")
+    def test_permission_error_to_remove_resource_metadata_file(self, remove):
+
+        entry_point = os.path.join(TEST_PACK_PATH, "actions", "test_entry_point.py")
+        metadata_file = os.path.join(TEST_PACK_PATH, "actions", "test_metadata.yaml")
+
+        remove.side_effect = PermissionError("No permission to delete file from disk")
+
+        # asserting metadata files exists
+        self.assertTrue(os.path.exists(metadata_file))
+
+        # asserting delete_action_files_from_pack function raises PermissionError
+        with self.assertRaises(PermissionError):
+            delete_action_files_from_pack(TEST_PACK, entry_point, metadata_file)
```

Comment:
How do we know this is the error for the metadata file?

Reply:
Please add an assert for the message of the mocked `PermissionError`.

Reply:
This one still needs to check either `PermissionError` or `ResourceDiskFilesRemovalError` or both. `Exception` is too broad.

Reply:
@cognifloyd I have added separate unit tests for `PermissionError`. As most probably this will be the exception will occur here and I'm raising same. But if any other exception occurs then I have used generalized `Exception` and raising `ResourceDiskFilesRemovalError`.

Reply:
Oh. I understand the test now. Cool.